packages/cloud_firestore/cloud_firestore_platform_interface/lib/cloud_firestore_platform_interface.dart

```diff
@@ -0,0 +1,219 @@
+
+  static CloudFirestorePlatform _instance = MethodChannelCloudFirestore();
+
+  // TODO(amirh): Extract common platform interface logic.
+  // https://github.com/flutter/flutter/issues/43368
+  static set instance(CloudFirestorePlatform instance) {
+    if (!instance.isMock) {
+      try {
+        instance._verifyProvidesDefaultImplementations();
+      } on NoSuchMethodError catch (_) {
+        throw AssertionError(
+            'Platform interfaces must not be implemented with `implements`');
+      }
+    }
+    _instance = instance;
+  }
+
+  /// This method ensures that [CloudFirestorePlatform] isn't implemented with `implements`.
+  ///
+  /// See class docs for more details on why using `implements` to implement
+  /// [CloudFirestorePlatform] is forbidden.
+  ///
+  /// This private method is called by the [instance] setter, which should fail
+  /// if the provided instance is a class implemented with `implements`.
+  void _verifyProvidesDefaultImplementations() {}
+
+  // Actual API 
+  // Global
+  /// Removes any listener by its handle.
+  /// All handles must be unique across al types of listeners.
+  Future<void> removeListener(int handle) async {
+    throw UnimplementedError('removeListener() is not implemented');
+  }
+
+  // Firestore
+  Future<void> enablePersistence(String app, {@required bool enable}) async {
+    throw UnimplementedError('enablePersistence() is not implemented');
+  }
+
+  Future<void> settings(String app, {
+      bool persistenceEnabled,
+      String host,
+      bool sslEnabled,
+      int cacheSizeBytes,
+    }) async {
+    throw UnimplementedError('settings() is not implemented');
+  }
+
+  Future<Map<String, dynamic>> runTransaction(String app, {
+    @required PlatformTransactionHandler transactionHandler,
+    int transactionTimeout,
+  }) async {
+    throw UnimplementedError('runTransaction() is not implemented');
+  }
+
+  // Document Reference
+  Future<void> setDocumentReferenceData(String app, {
+    @required String path,
+    Map<String, dynamic> data,
+    // TODO: Type https://firebase.google.com/docs/reference/js/firebase.firestore.SetOptions.html 
+    Map<String, dynamic> options,
+  }) async {
+    throw UnimplementedError('setDocumentReferenceData() is not implemented');
+  }
+
+  Future<void> updateDocumentReferenceData(String app, {
+    @required String path,
+    Map<String, dynamic> data,
+  }) async {
+    throw UnimplementedError('updateDocumentReferenceData() is not implemented');
+  }
+
+  // TODO: Type this return
+  Future<Map<String, dynamic>> getDocumentReference(String app, {
+    @required String path,
+    @required String source,
+  }) async {
+    throw UnimplementedError('getDocumentReference() is not implemented');
+  }
+
+  Future<void> deleteDocumentReference(String app, {
+    @required String path,
+  }) async {
+    throw UnimplementedError('deleteDocumentReference() is not implemented');
+  }
+
+  // TODO: Port to stream
```

Comment:
```suggestion
  // TODO(ditman): Port to stream
```

Reply:
(Yes, I'll get all the unnamed TODOs done before finalizing the CR, in particular this is already done! Yay!)